test/product-test.js

```diff
@@ -0,0 +1,299 @@
+          }
+        ],
+        model: '1l',
+        name: 'Fanta Limon 1l',
+        weight: {
+          value: 1.1,
+          unitcode: 'KGM',
+          unitText: 'kg'
+        }
+      })
+      .expect('Content-type', /json/)
+      .set('Accept', 'application/json')
+      .expect(201)
+      .end((err, res) => {
+        res.status.should.equal(201);
+        server
+        .get('/products')
+        .expect('Content-type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          res.status.should.equal(200);
+          res.body.length.should.be.eql(4);
+          done();
+        });
+      });
+    });
+
+    /* it("should return an 404 error",function(done){
+      server
+      .post("/products")
+      .send({
+        "name": "Fanta Limon 1l",
+        "ean": "5449033017888",
+        "description": "Boisson citronné avec des extraits de plantes. Rafraîchissant et vivifiant",
+        "category": "Drink",
+        "poids": "0.9"
+      })
+      .expect("Content-type",/json/)
+      .set('Accept', 'application/json')
+      .expect(404)
+      .end(function(err,res){
+        res.status.should.equal(404);
+        done();
+      });
+    });*/
+
+  });
+
+
+  describe('Remove', () => {
+    it('should remove a single product', (done) => {
+      server
+      .delete('/products')
+      .query({ ean: '5449000017888' })
+      .set('Accept', 'application/json')
+      .expect(204)
+      .end((err, res) => {
+        server
+        .get('/products')
+        .expect('Content-type', /json/)
+        .expect(200)
+        .end((err, res) => {
+          res.status.should.equal(200);
+          res.body.length.should.be.eql(2);
+          done();
+        });
+      });
+    });
+
+    it('should return a 400 cause of string ean', (done) => {
+      server
+      .delete('/products')
+      .query({ ean: 'dsfsdfd' })
+      .set('Accept', 'application/json')
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return a 400 cause of too short ean', (done) => {
+      server
+      .delete('/products')
+      .query({ ean: '123' })
+      .set('Accept', 'application/json')
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return a 400 cause of too long ean', (done) => {
+      server
+      .delete('/products')
+      .query({ ean: '123456798132456798' })
+      .set('Accept', 'application/json')
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
+    it('should return a 400 cause of not ean', (done) => {
+      server
+      .delete('/products')
+      .query({})
+      .set('Accept', 'application/json')
+      .expect(400)
+      .end((err, res) => {
+        res.status.should.equal(400);
+        done();
+      });
+    });
+
```

Comment:
In most of your tests you're writing:
```
.expect(value)
.end((......) => {
   ... should.equal(value)
}
```

After a quick check on supertest, we don't understand why you should check again the returned HTTP code.